### PR TITLE
After searching, the RHS should clear of anything

### PR DESCRIPTION
**Pull request by wtekeu**
This applies to search on both the People tab and on the Reports tab. We also reset the location so that the user can select the same item that was previously selected.

medic/medic-webapp#4825

### CODE REVIEW

**Comment by wtekeu** [comment]
Updated the code and the tests.

Thanks.

File: webapp/src/js/controllers/contacts.js [hunk 1]
```diff
@@ -253,13 +253,19 @@ var _ = require('underscore'),
...
       LiveList.contacts.clearSelected();
       LiveList['contact-search'].clearSelected();
+    };
+
+    $scope.$on('ClearSelected', function() {
+      clearSelection();
     });
 
     $scope.search = function() {
+      $state.go('contacts', null, { notify: false });
```
Comment (garethbowen) [inline webapp/src/js/controllers/contacts.js]: Use the route `contacts.detail` instead and the tests and UI will magically work.

File: webapp/src/js/controllers/reports.js [hunk 2]
```diff
@@ -255,6 +255,8 @@ angular
     };
 
     $scope.search = function() {
+      $state.go('reports', null, { notify: false });
```
Comment (garethbowen) [inline webapp/src/js/controllers/reports.js]: As above, use `reports.detail`

**Comment by wtekeu** [comment]
it worked! But I needed to explicitly set id to null:

$state.go('reports.detail', { id: null }, { notify: false });

**Pull request merged**